### PR TITLE
fix : 에러처리 일원화

### PR DESCRIPTION
**Pull request by GEUMDOHYEON**
## #️⃣ Part

- [ ] FE
- [x] BE

## 📝 작업 내용

> 에러 처리 일원화 + 에러처리 한글 패치 완료 
추가로 에러 문구 이상한 부분있으면 알려주세요. 바로 수정하겠습니다.

### CODE REVIEW

File: backend/src/challenges/challenges.controller.ts [hunk 1]
```diff
@@ -38,15 +38,15 @@ export class ChallengesController {
   ): Promise<ChallengeResponseDto> {
     const challenge = await this.challengeService.getChallengeInfo(challengeId);
     if (!challenge) {
-      throw new NotFoundException(`Challenge with ID ${challengeId} not found`);
+      throw new NotFoundException(
+        `${challengeId}의 챌린지를 찾을 수 없습니다.`,
```
Comment (pongpongie) [inline backend/src/challenges/challenges.controller.ts]: 이거 챌린지 아이디 굳이 알려줄 필요 없지 않나?

File: backend/src/challenges/challenges.service.ts [hunk 2]
```diff
@@ -94,12 +94,12 @@ export class ChallengesService {
 
     if (!editChall) {
       throw new NotFoundException(
-        `Challenge with ID ${challenge.challengeId} not found`,
+        `챌린지 ID ${challenge.challengeId}를 찾을 수 없습니다.`,
```
Comment (pongpongie) [inline backend/src/challenges/challenges.service.ts]: 프론트에서 아마 이거 바로 보여줄건데 굳이 챌린지 아이디 보여줄 필요 없을것 같긴해요

File: backend/src/challenges/challenges.service.ts [hunk 3]
```diff
@@ -412,7 +391,7 @@ export class ChallengesService {
 
     if (!nowAttendance) {
       throw new HttpException(
-        'Attendance does not exist',
+        '참여가 존재하지 않습니다.',
```
Comment (pongpongie) [inline backend/src/challenges/challenges.service.ts]: 이건 언제 사용하는거죠?

File: backend/src/users/users.service.ts [hunk 4]
```diff
@@ -448,21 +458,20 @@ export class UserService {
...
-  async changeTmpPassword(email: string) {
-    const user = await this.userRepository.findOne({
-      where: { email: email },
-    });
-
-    if (!user) {
-      return null;
+  checkAffirmation(affirmation: string) {
+    if (affirmation === '') {
+      throw new BadRequestException('확언 값이 없습니다.');
```
Comment (pongpongie) [inline backend/src/users/users.service.ts]: 확인 값이 뭐지여?

**Comment by pongpongie** [comment]
전체적으로 challengeId나 userId를 포함해서 에러 던지는 것이 맞는건지 저도 잘 모르겠어여!

**Comment by pongpongie** [comment]
This pull request focuses on improving error handling and removing unnecessary console logs in various backend service files. The most important changes include the introduction of more descriptive error messages, the use of appropriate exception classes, and the removal of console logs for better code clarity.

Improvements to error handling:

* [`backend/src/attendance/attendance.service.ts`](diffhunk://#diff-dbe42e0ee0b7cefdbe60f653482c7939e917c035660d6b5e2ed93acfaef36e6cL1-R1): Added `InternalServerErrorException` for better error reporting when saving attendance fails. [[1]](diffhunk://#diff-dbe42e0ee0b7cefdbe60f653482c7939e917c035660d6b5e2ed93acfaef36e6cL1-R1) [[2]](diffhunk://#diff-dbe42e0ee0b7cefdbe60f653482c7939e917c035660d6b5e2ed93acfaef36e6cL26-R28)
* [`backend/src/auth/auth.controller.ts`](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L20): Replaced generic error messages with more descriptive ones in Korean, and used appropriate exception classes such as `UnauthorizedException` and `HttpException`. [[1]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L20) [[2]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L34-L39) [[3]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L52-R58) [[4]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L81-L82) [[5]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L117-R110) [[6]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L135-L137) [[7]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L152-L170) [[8]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L189-R170)
* [`backend/src/auth/auth.service.ts`](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398L31-R33): Improved error messages during token validation and refresh processes. [[1]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398L31-R33) [[2]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398L49) [[3]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398L60-R64) [[4]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398L80-R92) [[5]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398L109)
* [`backend/src/challenges/challenges.controller.ts`](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L41-L49): Enhanced error messages for challenge-related operations and removed unnecessary console logs. [[1]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L41-L49) [[2]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L91) [[3]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L100) [[4]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L111) [[5]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L143-R145) [[6]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L168) [[7]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L180-R175) [[8]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L206-L219) [[9]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L239-R242)
* [`backend/src/challenges/challenges.service.ts`](diffhunk://#diff-6449019d3bb60ba3cabdcc1249a1086aef952c8b38fecd20aaae3057b8549334L97-R102): Updated error messages to be more descriptive and localized.

Code cleanup:

* Removed numerous `console.log` statements across multiple files to clean up the codebase and reduce clutter. [[1]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L34-L39) [[2]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L81-L82) [[3]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L135-L137) [[4]](diffhunk://#diff-57dfc3198cff39a2a2c4bd2e457609fe83e5287ede4110e2fa98b00753137e32L189-R170) [[5]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398L31-R33) [[6]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398L49) [[7]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398L60-R64) [[8]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398L109) [[9]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L41-L49) [[10]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L91) [[11]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L100) [[12]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L111) [[13]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L168) [[14]](diffhunk://#diff-586c9177dee73f018097204b905a27c2b062250a89f0d3fe462079330217db69L239-R242) [[15]](diffhunk://#diff-8e45ea36c88bea9d1bef49af4a7bc21c06ad89b9afdf1cb4d32fb658b415af0aL39-R40) [[16]](diffhunk://#diff-eaf9f1ec20a72c1f8a46faab7392811cefa51c047f14ece6bb607e87abc53411L6) [[17]](diffhunk://#diff-eaf9f1ec20a72c1f8a46faab7392811cefa51c047f14ece6bb607e87abc53411L18-L19) [[18]](diffhunk://#diff-eaf9f1ec20a72c1f8a46faab7392811cefa51c047f14ece6bb607e87abc53411L31-L45)

File: backend/src/users/users.service.ts (hunk 4 above)
Comment (GEUMDOHYEON) [reply backend/src/users/users.service.ts]: 오늘의 다짐 문구를 입력해주세요

Comment (GEUMDOHYEON) [reply backend/src/users/users.service.ts]: 유효성 체크도 추가

**Comment by GEUMDOHYEON** [comment]
챌린지 아이디나 유저 아이디 에러 처리 문구에서 삭제

File: backend/src/auth/auth.service.ts [hunk 5]
```diff
@@ -77,24 +73,23 @@ export class AuthService {
...
+        );
       }
     }
   }
 
   async generateAccessToken(userDto: UserDto): Promise<string> {
     const userFind = await this.userService.findUser(userDto.email);
     if (!userFind || userFind.password != userDto.password) {
-      throw new UnauthorizedException();
+      throw new UnauthorizedException('엑세스 토큰 생성 실패하였습니다.');
```
Comment (won0845) [inline backend/src/auth/auth.service.ts]: 액세스

File: backend/src/challenges/challenges.service.ts [hunk 6]
```diff
@@ -553,7 +531,7 @@ export class ChallengesService {
 
     if (startDateTime <= currentDate) {
       throw new BadRequestException(
-        'The start date and wake time must be in the future.',
+        '시작 날짜와 기상 시간이 유효하지 않습니다.',
```
Comment (won0845) [inline backend/src/challenges/challenges.service.ts]: "시작 날짜와 기상 시간이 유효하지 않습니다. "보다 챌린지 시작날짜는 현재시간보다 이후여야한다는 내용이 더 알맞을 듯

**Review by won0845 — APPROVED** [review]

**Review by pongpongie — APPROVED** [review]

File: backend/src/users/users.service.ts [hunk 7]
```diff
@@ -163,16 +177,20 @@ export class UserService {
...
         affirmation: affirmation,
       },
     );
-    console.log(result);
-    return result;
+    if (result.affected === 1) {
+      return true;
+    } else {
+      throw new InternalServerErrorException(
+        '확언 업데이트 중 문제가 발생했습니다.',
```
Comment (ifindary) [inline backend/src/users/users.service.ts]: 여기도 '확언'->'오늘의 다짐'으로 수정되어야 할 것 같습니다!

'오늘의 다짐 수정 중 문제가 발생했습니다'

File: backend/src/auth/auth.controller.ts [hunk 8]
```diff
@@ -149,25 +132,22 @@ export class AuthController {
...
       const result = await this.authService.refresh(refreshTokenDto);
-      console.log(result);
       if (result) {
         return {
           accessToken: result.accessToken,
           userId: result.userId,
         };
       } else {
-        throw new UnauthorizedException('Access Token create fail');
+        throw new UnauthorizedException('Access Token 생성 실패하였습니다.');
```
Comment (ifindary) [inline backend/src/auth/auth.controller.ts]: 액세스 토큰, 리프레쉬 토큰으로 용어 통일하면 좋을 것 같습니다!

**Pull request merged**